### ISSUE TITLE
Page number, line number, double spacing

### ISSUE THREAD

**Issue by adebali** [manubot/rootstock#244]
Most journals have some requirements like page number, continuous line number and double spacing. Is it possible to add these features for pdf export?

**Comment by dhimmel** [comment]
> If you mean line numbering for just regular paragraphs, whose wrapping is dynamic and can change, that's almost surely not possible, even with javascript.

Yes, it would be like [in this pdf](https://www.biorxiv.org/content/biorxiv/early/2014/12/11/011569.full.pdf). I guess the idea would be that the numbers get added after the dynamic line wrapping has occurred.

**Comment by adebali** [comment]
A confirmed straightforward way of page numbering:

```
pspdftool 'number(x=-1pt,y=-1pt,start=1,size=10)' output/manuscript.pdf output/manuscript_num.pdf
```

**Comment by dhimmel** [comment]
> A straightforward way of page numbering: But this method breaks all links

Ah too bad. I like the approach. pspdftool appears to be stagnant (no [updates](https://sourceforge.net/projects/pspdftool/)) since 2015), so perhaps right approach but wrong tool?

**Comment by adebali** [comment]
Another workaround with three dependencies: `pdftk`, `ps2pdf` and `enscript`.

* Create a bash file named `build/pagenum.sh`
* Content of the bash file:
```
#!/bin/bash
input="$1"
output="${1%.pdf}-num.pdf"
pagenum=$(pdftk "$input" dump_data | grep "NumberOfPages" | cut -d":" -f2)
enscript -L1 --header='|| $%/$=' --output - < <(for i in $(seq "$pagenum"); do echo; done) | ps2pdf - | pdftk "$input" multistamp - output $output
```
* Add the following line in the `build.sh`
```
bash build/pagenum.sh output/manuscript.pdf
```
The page-numbered output file will be `output/manuscript-num.pdf`.

It is kind of ugly that page numbers are in the header. I am sure this can be improved.

**Comment by lubianat** [comment]
Hello @adebali   and @dhimmel , any updates on that? Or is the pdftk + ps2pdf + enscript the current best approach?

**Comment by agitter** [comment]
@lubianat no one has been actively working on this, so the pdftk + ps2pdf + enscript approach is still the best I'm aware of.  We welcome improved suggestions if you or anyone else have ideas.

**Comment by lubianat** [comment]
Thanks, @agitter . I ended up compiling the docx via Manubot and adding line and page numbers in LibreOffice ([here is the docx](https://github.com/lubianat/technotype/blob/biorxiv_1/technotype_to_biorxiv.docx), by the way).

It is not perfect, but also not a big trouble. Way better than writing it all in Word or dueling with latex compilation. 

I wonder if it is possible to do LibreOffice macros. Maybe that would be a semi-automatic option.
